statsmodels/tsa/vector_ar/var_model.py

```diff
@@ -1611,12 +1612,16 @@ def irf_resim(self, orth=False, repl=1000, T=10,
         # df_model = self.df_model
         nobs = self.nobs
 
-        ma_coll = np.zeros((repl, T+1, neqs, neqs))
+        ma_coll = np.zeros((repl, horizon + 1, neqs, neqs))
 
-        def fill_coll(sim):
-            ret = VAR(sim, exog=self.exog).fit(maxlags=k_ar, trend=self.trend)
-            ret = ret.orth_ma_rep(maxn=T) if orth else ret.ma_rep(maxn=T)
-            return ret.cumsum(axis=0) if cum else ret
+        def fill_coll(simulation):
+            ret = VAR(simulation, exog=self.exog).fit(maxlags=k_ar,
+                                                      trend=self.trend)
+            if orth:
+                rep = ret.orth_ma_rep(maxn=horizon)
```

Comment:
pre-existing bug: orth_ma_rep is missing `p` (var_ or cov_decomp) option for generalized impulse response function

(Note, I only did some basic checking of the simulated methods for impulse response function when finishing up VECM merge. Not enough unit tests)

Reply:
p is optional now as it was supposed to be. 

https://github.com/statsmodels/statsmodels/pull/5220/files#diff-e2d6024b9d0ed6c0410bf2fffc1f1254R1169